core/src/main/scala/org/broadinstitute/dsp/workbench/welder/model.scala

```diff
@@ -45,11 +46,42 @@ object SyncMode {
 
 final case class BlobPath(asString: String) extends AnyVal
 
-sealed abstract class SourceUri
+sealed abstract class CloudProvider extends Product with Serializable {
+  def asString: String
+}
+object CloudProvider {
+
+  final case object Gcp extends CloudProvider {
+    override val asString = "GCP"
+  }
+
+  final case object Azure extends CloudProvider {
+    override val asString = "AZURE"
+  }
+
+  final case object None extends CloudProvider {
```

Comment:
what is this for? can we remove this?

Reply:
This is needed for the sourceUri contract, as a sourceURI must have a cloudProvider. A DataURI is a type of sourceURI that does not explicitly have a cloud provider, so we need this enum value to ensure consumers of a `SourceURI` handle those that are not azure/gcp specific properly.

Reply:
feels like we probably can model things a bit better tho...It doesn't feel right to add an arbitrary case becuz call site of `CloudProvider` has a specific use case to satisfy.

But we can leave this as is for this PR